packages/devextreme/testing/tests/DevExpress.ui.widgets/contextMenu.tests.js

```diff
@@ -569,6 +570,76 @@ QUnit.module('Rendering Scrollable', moduleConfig, () => {
 
         assert.roughEqual($scrollableContent.position().top, 0, 1, 'scroll position is reset');
     });
+
+    QUnit.test('Submenu height should be recalculated on dimension changes', function(assert) {
+        const instance = new ContextMenu(this.$element, {
+            items: [
+                { text: 1, items: [{ text: 11 }] },
+            ],
+            focusStateEnabled: true,
+            visible: true,
+            showSubmenuMode: { name: 'onHover', delay: 0 }
+        });
+        const $itemsContainer = instance.itemsContainer();
+        const item1 = $itemsContainer.find(`.${DX_MENU_ITEM_CLASS}`).first();
+
+        $(item1).trigger('dxclick');
+
+        const $nestedSubmenu = $(`.${DX_SUBMENU_CLASS}`).eq(1);
+        const spy = sinon.spy();
+
+        instance._setSubMenuHeight = spy;
+
+        resizeCallbacks.fire();
+
+        const args = spy.args[0][0];
+
+        assert.strictEqual(args[0], $nestedSubmenu[0], 'height recalculation is called for the nested submenu');
+        assert.strictEqual(spy.callCount, 1, 'no other recalculations');
+    });
+
+    QUnit.module('on dimension changed', {
+        setWindowHeight: function(windowHeight) {
+            implementationsMap.getOuterHeight = (el, ...args) => {
+                if(el === window) {
+                    return windowHeight;
+                }
+                return this._originalGetOuterHeight(el, ...args);
+            };
+        },
+        beforeEach: function() {
+            this._originalGetOuterHeight = implementationsMap.getOuterHeight;
+        },
+        afterEach: function() {
+            implementationsMap.getOuterHeight = this._originalGetOuterHeight;
+        }
+    }, () => {
+        QUnit.test('Submenu height should be recalculated', function(assert) {
+            const instance = new ContextMenu(this.$element, {
+                items: [{ text: 1, items: (new Array(99)).fill(null).map((_, idx) => ({ text: `item ${idx}` })) }],
+                focusStateEnabled: true,
+                visible: true,
+                showSubmenuMode: { name: 'onHover', delay: 0 }
+            });
+            const $itemsContainer = instance.itemsContainer();
+            const $item1 = $($itemsContainer.find(`.${DX_MENU_ITEM_CLASS}`).eq(0));
+
+            $item1.trigger('dxclick');
+
+            const windowHeight = 100;
+            this.setWindowHeight(windowHeight);
```

Comment:
Please, explore the idea and rewrite your tests + added tests for missed scenarios (flip, scroll, ...)